### ISSUE TITLE
.net core installer giving error on private agent.

### ISSUE THREAD

**Issue by garthbf** [Microsoft/azure-pipelines-tasks#8463]
## Troubleshooting
We are using private agents running server 2012 r2.
We know that we are behind a firewall and bluecoat. 
the task fails trying to get to the Json fro the .net core versions. 
We can get to that file in the web browser logged on directly to the server with the account that is running the agent service. 
the error is:
Unexpected token < in JSON at position 0

This error is given regardless of the version we try to install.

We need help to determine how this is connecting to github to get this file and what ports we need to open.

error log:

018-09-27T18:54:10.1683339Z ##[debug]Evaluating condition for step: 'Use .NET Core sdk 1.1.2'
2018-09-27T18:54:10.1684886Z ##[debug]Evaluating: succeeded()
2018-09-27T18:54:10.1685285Z ##[debug]Evaluating succeeded:
2018-09-27T18:54:10.1685813Z ##[debug]=> True
2018-09-27T18:54:10.1686410Z ##[debug]Result: True
2018-09-27T18:54:10.1687010Z ##[section]Starting: Use .NET Core sdk 1.1.2
2018-09-27T18:54:10.1693559Z ==============================================================================
2018-09-27T18:54:10.1693819Z Task         : .NET Core Tool Installer
2018-09-27T18:54:10.1694128Z Description  : Acquires a specific version of .NET Core from internet or the tools cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks.
2018-09-27T18:54:10.1694405Z Version      : 0.1.13
2018-09-27T18:54:10.1694598Z Author       : Microsoft Corporation
2018-09-27T18:54:10.1694841Z Help         : [More Information](https://go.microsoft.com/fwlink/?linkid=853651)
2018-09-27T18:54:10.1695084Z ==============================================================================
2018-09-27T18:54:10.6638732Z ##[debug]agent.workFolder=D:\agent-pbc2\_work
2018-09-27T18:54:10.6670406Z ##[debug]loading inputs and endpoints
2018-09-27T18:54:10.6676446Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2018-09-27T18:54:10.6692124Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2018-09-27T18:54:10.6696345Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2018-09-27T18:54:10.6699872Z ##[debug]loading INPUT_PACKAGETYPE
2018-09-27T18:54:10.6702386Z ##[debug]loading INPUT_VERSION
2018-09-27T18:54:10.6708401Z ##[debug]loaded 5
2018-09-27T18:54:10.7258211Z ##[debug]Agent.ProxyUrl=undefined
2018-09-27T18:54:10.7261656Z ##[debug]check path : D:\agent-pbc2\_work\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.1.13\node_modules\vsts-task-tool-lib\lib.json
2018-09-27T18:54:10.7264240Z ##[debug]adding resource file: D:\agent-pbc2\_work\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.1.13\node_modules\vsts-task-tool-lib\lib.json
2018-09-27T18:54:10.7273198Z ##[debug]system.culture=en-US
2018-09-27T18:54:10.7317103Z ##[debug]Setting resource path to D:\agent-pbc2\_work\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.1.13\task.json
2018-09-27T18:54:10.7320011Z ##[debug]check path : D:\agent-pbc2\_work\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.1.13\task.json
2018-09-27T18:54:10.7321902Z ##[debug]set resource file to: D:\agent-pbc2\_work\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.1.13\task.json
2018-09-27T18:54:10.7322462Z ##[debug]system.culture=en-US
2018-09-27T18:54:10.7385244Z ##[debug]packageType=sdk
2018-09-27T18:54:10.7388800Z ##[debug]version=1.1.2
2018-09-27T18:54:10.7427802Z Tool to install: .NET Core sdk version 1.1.2.
2018-09-27T18:54:10.7431857Z ##[debug]isExplicit: 1.1.2
2018-09-27T18:54:10.7432481Z ##[debug]explicit? true
2018-09-27T18:54:10.7438942Z Checking if a cached copy exists for this version...
2018-09-27T18:54:10.7439397Z ##[debug]isExplicit: 1.1.2
2018-09-27T18:54:10.7440022Z ##[debug]explicit? true
2018-09-27T18:54:10.7441159Z ##[debug]Agent.Version=2.138.4
2018-09-27T18:54:10.7441516Z ##[debug]Agent.ToolsDirectory=D:\agent-pbc2\_work\_tool
2018-09-27T18:54:10.7442301Z ##[debug]checking cache: D:\agent-pbc2\_work\_tool\dncs\1.1.2\x64
2018-09-27T18:54:10.7443667Z ##[debug]not found
2018-09-27T18:54:10.7444059Z Cache does not contains this particular .NET Core. Will be downloaded and installed.
2018-09-27T18:54:10.7446521Z Detected platform (Primary): win-x64
2018-09-27T18:54:10.7446770Z Getting URL to download .NET Core sdk version: 1.1.2.
2018-09-27T18:54:10.7453753Z ##[debug]agent.proxyurl=undefined
2018-09-27T18:54:10.7454195Z ##[debug]AZURE_HTTP_USER_AGENT=VSTS_408d5e33-5a6f-43f5-9102-4268575e41fc_build_2747_0
2018-09-27T18:54:11.2675408Z ##[debug]task result: Failed
2018-09-27T18:54:11.2732282Z ##[error]Unexpected token < in JSON at position 0
2018-09-27T18:54:11.2745405Z ##[debug]Processed: ##vso[task.issue type=error;]Unexpected token < in JSON at position 0
2018-09-27T18:54:11.2760018Z ##[debug]Processed: ##vso[task.complete result=Failed;]Unexpected token < in JSON at position 0
2018-09-27T18:54:11.2782782Z ##[section]Finishing: Use .NET Core sdk 1.1.2

**Comment by bishal-pdMSFT** [comment]
@garthbf I have created a new issue https://github.com/dotnet/core/issues/1977. We will also be handling this from our side.

**Comment by bishal-pdMSFT** [comment]
@garthbf sorry for late reply. The task tries to download releases json file from https://github.com/dotnet/core/blob/master/release-notes/releases.json. It uses this file to get the download URL for a particular dotnet core version and a particular platform. 
You can allow this host through the firewall.

**Comment by amaljg** [comment]
@garthbf - Are you unblocked over here? Let us know if you still face any issues.

**Comment by kuvinodms** [comment]
@garthbf - I'm closing this issue. Feel free to reopen if you're still blocked.
